java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture6/tasks/_3/UserRepository.java

```diff
@@ -0,0 +1,31 @@
+package com.zoolatech.lecture6.tasks._3;
+
+import java.util.stream.Collectors;
+
+class UserRepository {
+    UserCache userCache;
+    UserTable userTable;
+
+    public UserRepository(UserCache userCache, UserTable userTable) {
+        this.userCache = userCache;
+        this.userTable = userTable;
+    }
+
+    public String findUserEmail(String id) throws UserMissingException {
+        if (userCache.findUser(id).isPresent()) {
```

Comment:
1. try to remove the code duplication
2. try to get rid of isPresent methods with a help of Optional's `or...` methods